File: modules/screenorientation/android/src/main/java/expo/modules/screenorientation/ScreenOrientationModule.kt
package expo.modules.screenorientation

import android.content.res.Configuration
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ScreenOrientationModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ScreenOrientation")

    Function("getOrientation") {
      val orientation = appContext.reactContext?.resources?.configuration?.orientation
      
      if (orientation == Configuration.ORIENTATION_LANDSCAPE) {
        "landscape"
      } else {
        "portrait"
      }
    }
  }
}

File: modules/screenorientation/ios/ScreenOrientationModule.swift
import ExpoModulesCore

public class ScreenOrientationModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ScreenOrientation")

    Function("getOrientation") { () -> String in
      let interfaceOrientation = UIApplication.shared.windows.first?.windowScene?.interfaceOrientation
      
      if interfaceOrientation?.isLandscape ?? false {
        return "landscape"
      } else {
        return "portrait"
      }
    }
  }
}